common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Jobs/Default/DefaultJobOrchestrator.cs

```diff
@@ -48,6 +48,9 @@ public class DefaultJobOrchestrator : IJobOrchestrator {
             string continuationToken = null;
             do {
                 var jobList = await _jobRepository.QueryAsync(query, continuationToken, null, ct);
+                if (jobList?.Jobs == null) {
+                    continue;
```

Comment:
We should also assign value to `continuationToken` here so that loop is broken when there are no more entries to retrieve.

```
ontinuationToken = jobList.ContinuationToken;
```

Reply:
agreed. actually, the continuationToken should be reset to null. done that.

Reply:
Can we actually exclude cases where `jobList.Jobs` would be `null` but it has a continuation token ? To be honest, that would be a weird scenario and wouldn't make much sense, but I am wondering if we can rely on that never happening. Also, if we set `continuationToken` to null, isn't that effectively a `break` instead of `continue` ?